apps/music/views/songs/view.js

```diff
@@ -8,47 +8,65 @@ var SongsView = View.extend(function SongsView() {
   this.searchResults = document.getElementById('search-results');
   this.list = document.getElementById('list');
 
-  this.searchBox.addEventListener('search', (evt) => this.search(evt.detail));
+  this.client.on('databaseChange', () => this.update());
+
+  this.configureList();
+  this.update();
+});
 
-  this.searchResults.addEventListener('open', () => {
+SongsView.prototype.configureSearch = function() {
+  var results = this.searchResults;
+  var box = this.searchBox;
+
+  box.addEventListener('search', (evt) => this.search(evt.detail));
+
+  results.addEventListener('open', () => {
     this.client.method('searchOpen');
   });
 
-  this.searchResults.addEventListener('close', () => {
+  results.addEventListener('close', () => {
     this.client.method('searchClose');
-    this.list.scrollTop = this.searchBox.HEIGHT;
+    this.list.scrollTop = box.HEIGHT;
   });
 
-  this.searchResults.addEventListener('resultclick', (evt) => {
+  results.addEventListener('resultclick', (evt) => {
     var link = evt.detail;
     if (link) {
       this.queueSong(link.dataset.filePath);
-
       this.client.method('navigate', link.getAttribute('href'));
     }
   });
 
   this.searchResults.getItemImageSrc = (item) => this.getThumbnail(item.name);
+};
 
+SongsView.prototype.configureList = function() {
+  var list = this.list;
+
+  // Scroll search out of view, even when
+  // there arent' enough list items to scroll.
   this.list.scrollTop = this.searchBox.HEIGHT;
   this.list.minScrollHeight = `calc(100% + ${this.searchBox.HEIGHT}px)`;
 
   this.list.configure({
-    getItemImageSrc: (item) => {
-      return this.getThumbnail(item.name);
-    }
+    getItemImageSrc: (item) => this.getThumbnail(item.name)
   });
 
-  this.list.addEventListener('click', (evt) => {
+  list.addEventListener('click', (evt) => {
     var link = evt.target.closest('a[data-file-path]');
     if (link) {
       this.queueSong(link.dataset.filePath);
     }
   });
 
-  this.client.on('databaseChange', () => this.update());
-  this.update();
-});
+  // Show the view only when list has something rendered,
+  // this prevents Gecko painting unnecessarily.
+  list.addEventListener('rendered', function fn() {
```

Comment:
See comment above about adding a `once()` utility method to the base view.js module.